include/baresip.h

```diff
@@ -1652,14 +1668,25 @@ void module_app_unload(void);
 
 int event_encode_dict(struct odict *od, struct ua *ua, enum ua_event ev,
 		      struct call *call, const char *prm);
+int odict_encode_event(struct odict *od, struct event *event);
 int event_add_au_jb_stat(struct odict *od_parent, const struct call *call);
 int  uag_event_register(ua_event_h *eh, void *arg);
 void uag_event_unregister(ua_event_h *eh);
+int  event_register(event_h *eh, void *arg);
+void event_unregister(event_h *eh);
 void ua_event(struct ua *ua, enum ua_event ev, struct call *call,
 	      const char *fmt, ...);
+int event_app_emit(enum ua_event ev, void *arg, const char *fmt, ...);
+int event_ua_emit(enum ua_event ev, struct ua *ua, const char *fmt, ...);
+int event_call_emit(enum ua_event ev, struct call *call, const char *fmt, ...);
+int event_sip_msg_emit(enum ua_event ev, const struct sip_msg *msg,
+		       const char *fmt, ...);
```

Comment:
is the `call` valid in this case ?

Reply:
Yes. Here the table from above extended with the function names:

| function name | event class | union field |
| --- | --- | --- |
| event_ua_emit | EVENT_CLASS_UA | event.u.ua |
| event_call_emit | EVENT_CLASS_CALL | event.u.call |
| event_app_emit | EVENT_CLASS_APP | event.u.arg |
| event_sip_msg_emit | EVENT_CLASS_SIP  |  event.u.msg |